[Analytics] Adds another GraphQL query (#15658)
cloudflare/cloudflare-docs: content/analytics/graphql-api/tutorials/end-customer-analytics/index.md

}
```

## Top 10 consuming URLs in a zone

We are going to use the GraphQL Analytics API to query the top 10 consuming URLs from a zone, helping you identify the URLs with the highest resource usage. Here are some configuration instructions:

> - To filter on a specific hostname, add the line `"clientRequestHTTPHost": "'$2'"`, below `"requestSource"`." 

Reviewer: ```suggestion
- To filter on a specific hostname, add the line `"clientRequestHTTPHost": "'$2'"` below `"requestSource"`." 
```

- Replace `API_TOKEN` with your generated API token using the `Read all resources` permissions. The script will only access zones available to the token's creator. 

> - Pass the zoneID as a parameter `ARG=$1`. 

Reviewer: ```suggestion
- Pass the zone ID as a parameter `ARG=$1`. 
```

Reviewer: Either you're talking about the "zone ID" (the ID of the zone) or you're mentioning the ``zoneID`` - the parameter. If it's the latter, please use backticks

> - To calculate the current date and the date from 30 days ago, use gdate on Mac:

Reviewer: ```suggestion
- To calculate the current date and the date from 30 days ago, use `gdate` on Mac:
```

  - `CURRENTDATE=$(gdate -u +'%FT%TZ')` 
  - `OLDDATE=$(gdate -d '-30 days' -u +'%FT%TZ')`. 
- For specific dates within the last 30 days, set `CURRENTDATE` and `OLDDATE` variables in the format `"YYYY-MM-DDTHH:MM:SSZ"`.

### API call

```bash

> curl -s -H "Authorization: Bearer $API_TOKEN" -H "Content-Type:application/json" \
> -XPOST https://api.cloudflare.com/client/v4/graphql \
> -d'{

Reviewer 2: ```suggestion
curl --silent \
https://api.cloudflare.com/client/v4/graphql \
--header "Authorization: Bearer $API_TOKEN" \
--header "Content-Type: application/json" \
--data '{
```

>   "query": "{\n  viewer {zones(filter: {zoneTag: $zoneTag}) {topPaths: httpRequestsAdaptiveGroups(filter: $filter, limit: 10, orderBy: [sum_edgeResponseBytes_DESC]) {count sum {edgeResponseBytes} dimensions {metric: clientRequestPath}}}}}",

Reviewer 3: ```suggestion
  "query": "{viewer {zones(filter: {zoneTag: $zoneTag}) {topPaths: httpRequestsAdaptiveGroups(filter: $filter, limit: 10, orderBy: [sum_edgeResponseBytes_DESC]) {count sum {edgeResponseBytes} dimensions {metric: clientRequestPath}}}}}",
```

  "variables": {
    "zoneTag": "'$ARG'",
    "filter": {
      "AND": [
        {
          "datetime_geq": "'$OLDDATE'",
          "datetime_leq": "'$CURRENTDATE'"
        },
        {
          "requestSource": "eyeball"
        }
      ]
    }
  }

> }' | jq -r 'try .data.viewer.zones[].topPaths[] | "\(.dimensions.metric) | \(.sum.edgeResponseBytes)"' | numfmt -d'|' --field=2 --to=iec-i

Reviewer 3: ```suggestion
}' | jq -r 'try .data.viewer.zones[].topPaths[] | "\"\(.dimensions.metric)\": \(.sum.edgeResponseBytes)"' | sort
```

numfmt is an overkill (convert numbers to human-readable strings) and not a standard tool, simple sorting should be enough to just illustrate the idea.

I also suggesting wrapping metric into double-quotes.

}
```

## Top 10 consuming URLs in a zone

We are going to use the GraphQL Analytics API to query the top 10 consuming URLs from a zone, helping you identify the URLs with the highest resource usage. Here are some configuration instructions:

- To filter on a specific hostname, add the line `"clientRequestHTTPHost": "'$2'"` below `"requestSource"`." 
- Replace `API_TOKEN` with your generated API token using the `Read all resources` permissions. The script will only access zones available to the token's creator. 

> - Pass the zone ID as a parameter `ARG=$1`. 

Author: ```suggestion
- Pass the zone ID (`zoneTag`) as a parameter `ARG=$1`. 
```
